internal/internal_workflow_client.go

```diff
@@ -1147,8 +1147,31 @@ func (wc *WorkflowClient) OperatorService() operatorservice.OperatorServiceClien
 	return operatorservice.NewOperatorServiceClient(wc.conn)
 }
 
+type loadCapabilitiesConfig struct {
+	getSystemInfoTimeout time.Duration
+}
+
+var defaultLoadCapabilitiesOptions = loadCapabilitiesConfig{
+	getSystemInfoTimeout: getSystemInfoTimeout,
+}
+
+type loadCapabilitiesOption func(config *loadCapabilitiesConfig)
+
+func withGetSystemInfoTimeout(timeout time.Duration) loadCapabilitiesOption {
+	return func(config *loadCapabilitiesConfig) {
+		if timeout > 0 {
+			config.getSystemInfoTimeout = timeout
+		}
+	}
+}
+
 // Get capabilities, lazily fetching from server if not already obtained.
-func (wc *WorkflowClient) loadCapabilities() (*workflowservice.GetSystemInfoResponse_Capabilities, error) {
+func (wc *WorkflowClient) loadCapabilities(ctx context.Context, opts ...loadCapabilitiesOption) (*workflowservice.GetSystemInfoResponse_Capabilities, error) {
```

Comment:
I think you're overcomplicating a non-exported function with this new option concept. Just pass in the timeout as a param and change the global val name to `defaultGetSystemInfoTimeout` and use the default if the passed in one is 0.